services/httpd/response_writer.go

```diff
@@ -124,19 +129,25 @@ type jsonFormatter struct {
 
 func (f *jsonFormatter) WriteResponse(w io.Writer, resp Response) (err error) {
 	var b []byte
+
 	if f.Pretty {
 		b, err = json.MarshalIndent(resp, "", "    ")
 	} else {
 		b, err = json.Marshal(resp)
 	}
 
 	if err != nil {
-		_, err = io.WriteString(w, err.Error())
-	} else {
-		_, err = w.Write(b)
+		err = unnestError(err)
```

Comment:
Need to format this error as json (e.g., `{"error":"json: unsupported value: +Inf"}`) and write to the client.

Reply:
The error is correctly formatted by the JSON marshaller.  The top level error looks like:

{"error":"json: error calling MarshalJSON for type httpd.Response: json: error calling MarshalJSON for type *query.Result: json: unsupported value: +Inf"}

The deepest error after unnesting is this:

{"error":"json: unsupported value: +Inf"}

Reply:
The write to the client happens in the caller, (*responseWriter).WriteResponse(), rather than in (*jsonFormatter).WriteResponse() in the previous version.